Marlin/src/feature/tmc_util.h

```diff
@@ -393,34 +393,16 @@ void test_tmc_connection(const bool test_x, const bool test_y, const bool test_z
 
     template<class TMC, char AXIS_LETTER, char DRIVER_ID, AxisEnum AXIS_ID>
     bool TMCMarlin<TMC, AXIS_LETTER, DRIVER_ID, AXIS_ID>::test_stall_status() {
-      uint16_t sg_result = 0;
-
       this->switchCSpin(LOW);
 
-      if (this->TMC_SW_SPI != nullptr) {
-        this->TMC_SW_SPI->transfer(TMC2130_n::DRV_STATUS_t::address);
-        this->TMC_SW_SPI->transfer16(0);
-        // We only care about the last 10 bits
-        sg_result = this->TMC_SW_SPI->transfer(0);
-        sg_result <<= 8;
-        sg_result |= this->TMC_SW_SPI->transfer(0);
-      }
-      else {
-        SPI.beginTransaction(SPISettings(16000000/8, MSBFIRST, SPI_MODE3));
-        // Read DRV_STATUS
-        SPI.transfer(TMC2130_n::DRV_STATUS_t::address);
-        SPI.transfer16(0);
-        // We only care about the last 10 bits
-        sg_result = SPI.transfer(0);
-        sg_result <<= 8;
-        sg_result |= SPI.transfer(0);
-        SPI.endTransaction();
-      }
+      // read stallGuard flag from TMC library, will handle HW and SW SPI
+      TMC2130_n::DRV_STATUS_t drv_status{0};
+      drv_status.sr = this->DRV_STATUS();
+
```

Comment:
@teemuatlut — Is this correct…? This much (legacy) code is replaced by a simple update of `drv_status.sr`…? I'd love to merge this asap if this is all correct.

Reply:
It's not legacy code. There's a specific reason I did it this way and it was to increase the speed of the checks.
Before merging this the status was checked after each read or in other words, after sending 40bits. Now with this you're sending 40 dummy bits with which you don't do anything and only after then you send another 40bits to shift out the response. Basically this cuts down the communication and reaction speed to half. Also before there were 2 digitalWrites and now there's 4.

This likely won't break anything but going through the code it doesn't achieve anything either, but it does make things slower.


The changes to `endstops.cpp` aren't necessary if you disabled the endstop pins with `-1`.

The old code might've been bugged by the `nullptr` check. The SW_SPI pointer by default is initialized to `NULL` by the library and these two may not the be same exact thing.

---

The changes can stay but with the acknowledgement of the downsides and that the original code wasn't done the way it was just to make it more complicated.

Reply:
> Basically this cuts down the communication and reaction speed to half. Also before there were 2 digitalWrites and now there's 4.

What are the practical consequences of this? In terms of command dispatch rate, for example.

Reply:
> The old code might've been bugged by the nullptr check. The SW_SPI pointer by default is initialized to NULL by the library and these two may not the be same exact thing.

I'm pretty sure that this is not the issue. I previously had my fix within the SW_SPI branch and it worked.

> The changes to endstops.cpp aren't necessary if you disabled the endstop pins with -1

If that is true I would vote to revert the changes in endstops.cpp in favor of documenting this behaviour in configuration_adv.h